app/models/res_tful.rb

```diff
@@ -0,0 +1,6 @@
+class ResTful < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+end
```

Comment:
- You have done well so far, but it seems to me that you have not properly installed devise in your project, please follow the guidelines [here](https://github.com/heartcombo/devise#getting-started) on how to properly install devise on your application. When devise is properly installed some view files will be automatically generated to let users sign in/sing_up on your app. Please fix this 👍 👍 
 
- Also please note that when you install devise properly, these modules for your database will automatically be generated inside your user model. Please do well to ensure that all modules required for this project are enabled and delete this file as it would not be needed. Please fix this🙏 
Please run this command to install devise in your user model: 
```ruby
rails generate devise User
```

- You might also find this [doc](https://hackernoon.com/using-devise-in-your-ruby-on-rails-application-a-step-by-step-guide-m92i3y5s) in helping you install devise properly. Please fix this 👍 👍

Reply:
- Good job @firstpj, but there are still some changes left to be made before your project can be approved, as mentioned in my previous review, please ensure that your devise view pages have been configured to allow new users to register on your application. Please follow this [guide](https://github.com/heartcombo/devise#configuring-views) to configure this 🙏 

- After configuring your view files, please make sure that only authenticated users can access any of your controller actions. Please follow this [guide](https://github.com/heartcombo/devise#controller-filters-and-helpers) to achieve this 😄 
In your application controller please remove the current_user method like I mentioned in my previous review, devise provides a helper for this. Please fix this 👍 👍